python/cudf/cudf/_lib/orc.pyx

```diff
@@ -341,11 +341,12 @@ cdef orc_reader_options make_orc_reader_options(
         orc_reader_options.builder(src)
         .stripes(strps)
         .skip_rows(skip_rows)
-        .num_rows(num_rows)
         .timestamp_type(data_type(timestamp_type))
         .use_index(use_index)
         .build()
     )
+    if num_rows >= 0:
+        opts.set_num_rows(num_rows)
```

Comment:
Why the switch to calling the function instead of using the option?

Reply:
Because `num_rows` is now stored as `std::optional`. When the `optional` is `nullopt`, it denotes that user did not pass this option.

Reply:
Because we don't want to set the option when user did not pass a value. Python layer passes -1 as "not set" (same as what we used before I switched to std::optional), which is why we use the `>= 0`condition here.